docs/ts/develop/middleware.md

```diff
@@ -0,0 +1,95 @@
+---
+seotitle: Using Middleware in your backend application
+seodesc: See how you can use middleware in your backend application to handle cross-cutting generic functionality, like request logging, auth, or tracing.
+title: Middleware
+subtitle: Handling cross-cutting, generic functionality
+lang: lang
+---
+
+Middleware is a way to write reusable code that runs before or after (or both)
+the handling of API requests, often across several (or all) API endpoints.
+
+It's commonly used to implement cross-cutting concerns like
+[request logging](/docs/ts/observability/logging),
+[authentication](/docs/ts/develop/auth),
+[tracing](/docs/ts/observability/tracing),
+and so on. One of the benefits of Encore is that
+all of these use cases are already handled out-of-the-box, so there's no
+need to use middleware for those things.
+
+Nonetheless, there are several use cases where it can be useful to write
+reusable functionality that applies to multiple API endpoints, and middleware
+is a good solution in those cases.
```

Comment:
```suggestion
However, when developing applications there's often some use cases where it can be useful to write
reusable functionality that applies to multiple API endpoints, and middleware
is a good solution for this.
```